examples/fabric-admin/device_manager/DeviceSynchronization.cpp

```diff
@@ -205,8 +206,80 @@ void DeviceSynchronizer::StartDeviceSynchronization(chip::Controller::DeviceCont
     mCurrentDeviceData.has_is_icd = true;
     mCurrentDeviceData.is_icd     = deviceIsIcd;
 
-    mDeviceSyncInProcess = true;
-
+    ReturnOnFailure(controller->GetConnectedDevice(nodeId, &mOnDeviceConnectedCallback, &mOnDeviceConnectionFailureCallback));
     mController = controller;
-    controller->GetConnectedDevice(nodeId, &mOnDeviceConnectedCallback, &mOnDeviceConnectionFailureCallback);
+    MoveToState(State::Connecting);
+}
+
+void DeviceSynchronizer::GetUid(EndpointId remoteEndpointIdOfInterest)
+{
+    VerifyOrDie(mState == State::ReceivedResponse);
+    VerifyOrDie(mController);
+    VerifyOrDie(DeviceMgr().IsFabricSyncReady());
+    auto remoteBridgeNodeId = DeviceMgr().GetRemoteBridgeNodeId();
+
+    CHIP_ERROR err = mUidGetter.GetUid(
+        [this](std::optional<CharSpan> aUniqueId) {
+            if (aUniqueId.has_value())
+            {
+                this->mCurrentDeviceData.has_unique_id = true;
+                memcpy(this->mCurrentDeviceData.unique_id, aUniqueId.value().data(), aUniqueId.value().size());
+            }
```

Comment:
would the false branch represent an error (we request a UID when one is missing, yet we get nothing)? Should we log an error in this case?